SmartCar/Arduino/sketch_main/sketch_main.ino

```diff
@@ -242,6 +287,81 @@ if(temperatureActivation){
     }   
 }
 
+/*
+ * Updates the camera servos on each iteration, until a 0 is given.
+ * Checks both x and y individually so that they both can be moved at once.
+ * Author: Pontus Laestadius.
+ * Since: 25th of April 2017
+*/
+void cameraUpdate(int &x, int &y) {
+
+  if (x != 0) { 
+    setCameraServoRelative(camera_servo_x, X_MAX_ANGLE, X_MIN_ANGLE, x, cameraX);
+  }
+  if (y != 0) {
+    setCameraServoRelative(camera_servo_y, Y_MAX_ANGLE, Y_MIN_ANGLE, y, cameraY);
+  }
+
+}
+
+/*
+   Author: Pontus Laestadius
+   Since: 25th of April 2017
+   Purpose: Handles the camera's servos with constraits.
+   Do not call this method directly, Rather use setCameraServoRelative.
+*/
+int moveCamera(Servo &servo, const int &max, const int &min, int degrees) {
+
+  // If the valueReceivedFromTheRaspberrys exceed the angle the servo is able to turn.
+  if (degrees < min || degrees > max) {
+
+    // Reset to either the minimum or maximum depending on which one it exceeds.
+    servo.write(degrees < min ? min : max);
+
+    // End the method. And return the valueReceivedFromTheRaspberry the camera angle is now set to.
+    return degrees < min ? min : max;
+  }
+
+  // Turn the servo to the indicated angle.
+  servo.write(degrees);
+  return degrees;
+}
+
+/**
+   Author: Pontus Laestadius
+   Since: 25th of April 2017
+   Converts a valueReceivedFromTheRaspberry for the servo relative method.
+   Param: v is an integer. either 1 or 2 (everything else returns 0), that gets converted to -1 or 1.
+*/
+int setCameraConverter(int v) {
+  // If the method is called without any changed valueReceivedFromTheRaspberrys. End.
+  if (v == 1) {
+    return -v;
+  } else if (v == 2) {
+    return  --v;
+  }
+  return 0;
+}
+
+/**
+   Author: Pontus Laestadius
+   Since: 25th of April 2017
+   Moves the camera relative to the current position. Such as left, right, up and down.
+   Param: Servo to be moved, max and min angles the servo can move. valueReceivedFromTheRaspberry, either 1 for lowing the angle and 2 for increasing it.
+   Amplifier amplifies the input coordinates. Add functionality so this valueReceivedFromTheRaspberry can be changed remotely to change speed.
+*/
+
+void setCameraServoRelative(Servo &servo, const int &max, const int &min, int valueReceivedFromTheRaspberry, int &currentAngle) {
```

Comment:
You have to explain this further to me irl and probably add commenting so it's more understandable.